packages/docs-page/server/loaders/remote-content.ts

```diff
@@ -162,10 +162,6 @@ export default class RemoteContentLoader implements DataLoader {
 
     const versionMetadataList: VersionMetadataItem[] =
       await cachedFetchVersionMetadataList(this.opts.product)
-    // remove trailing index to ensure we fetch the right document from the DB
-    const pathParamsNoIndex = paramsNoVersion.filter(
-      (param, idx, arr) => !(param === 'index' && idx === arr.length - 1)
-    )
```

Comment:
I _think_ removing this gets us the desired behaviour. But, I feel like I might be missing additional context here. My assumption here:
- The only situation in which incoming URL `params` parts would have a trailing `index`, AND we want to resolve and render the page at that URL, is for pages that are genuinely called `index`.
    - And such pages should resolve from the content API _even with_ the trailing index in their `fullPath`